Kickstarter-iOS/ViewModels/AppDelegateViewModel.swift

```diff
@@ -1188,3 +1210,38 @@ private func configRetainingDebugFeatureFlags(_ config: Config) -> Config {
 
   return config |> Config.lens.features .~ currentFeatures.withAllValuesFrom(storedFeatures)
 }
+
+private func willHandleBrazeUrlAndNavigation(urlAndExtras: (URL?, [AnyHashable: Any]?)?)
+  -> (Bool, Navigation?) {
+  guard
+    let (maybeUrl, _) = urlAndExtras,
+    let url = maybeUrl
+  else { return (false, nil) }
+
+  guard let navigation = Navigation.match(url) else {
+    return (false, nil)
+  }
+
+  return (true, navigation)
+}
+
+private func updateUserNotificationSetting(navigation: Navigation) -> SignalProducer<User, Never> {
```

Comment:
Bit convoluted but this is in order to keep the enabling/disabling at least somewhat dynamic. This allows us to pass in say `notify_mobile_of_messages` or `notify_mobile_of_follower` and have it retrieve those settings from the current `User` in a dictionary form, update that dictionary, turn it back into `User.Notifications` and then lens over the `User` to update it.